include/fastdds/dds/topic/TypeSupport.hpp

```diff
@@ -227,6 +227,17 @@ class TypeSupport : public std::shared_ptr<fastdds::dds::TopicDataType>
         return !*this;
     }
 
+    /**
+     * Copy assignment operator
+     */
+    RTPS_DllAPI TypeSupport& operator =(
+            const TypeSupport& other)
+    {
+        Base::operator =(
+                other);
+        return *this;
+    }
```

Comment:
```suggestion
    RTPS_DllAPI TypeSupport& operator =(
            const TypeSupport& type) = default;
```

I think we can use the default copy assignment here.

I would also change the copy constructor to use also the default one.

```
RTPS_DllAPI TypeSupport(
        const TypeSupport& type) = default;
```